docs/user_manual.md

```diff
@@ -49,40 +60,277 @@ python -m virtualenv venv
+  and the current workaround forces this package to be build from sources
+  on most platforms
+  (see [#253](https://github.com/openvinotoolkit/datumaro/issues/253)).
+  If you need to use 2.0.2, make sure it is linked with the same version
+  of `numpy` as `TensorFlow` by reinstalling the package:
+  ``` bash
+  pip uninstall pycocotools
+  pip install pycocotools --no-binary=pycocotools
+  ```
+
+- When installing directly from the repository, you can change the
+  installation branch with `...@<branch_name>`. Also use `--force-reinstall`
+  parameter in this case. It can be useful for testing of unreleased
+  versions from GitHub pull requests.
+
+## How to use Datumaro
+
+As a standalone tool or a Python module:
 
 ``` bash
 datum --help
-```
-
-As a python module:
-> The directory containing Datumaro should be in the `PYTHONPATH`
-> environment variable or `cvat/datumaro/` should be the current directory.
 
-``` bash
 python -m datumaro --help
 python datumaro/ --help
 python datum.py --help
 ```
 
-As a python library:
+As a Python library:
 
 ``` python
-import datumaro
+from datumaro.components.project import Project
+from datumaro.components.dataset import Dataset
+from datumaro.components.extractor import Label, Bbox, DatasetItem
+...
+dataset = Dataset.import_from(path, format)
+...
+```
+
+### Glossary
+
+- Basic concepts:
+  - dataset - A number of media (dataset items) and associated annotations
+  - project - A combination of multiple datasets, plugins, models and metadata
+
+- Project versioning concepts:
+  - data source - a link to a dataset or a copy of a dataset inside a project.
+    Basically, an URL + dataset format name
+  - project revision - a commit hash or a named reference from
+    Git (branch, tag, HEAD~3 etc.).
+  - working / head / revision tree - a project build tree and plugins at
+    a specified revision
+  - data source revision - a data source hash at a specific stage
+  - object - a tree or a data source revision data
+
+- Dataset path concepts: <a id="revpath"></a>
+  - source / dataset / revision / project path - a path to a dataset in a
+    special format
+
+    - (project local) **rev**ision **path**s - a way to specify the path
+      to a source revision in the CLI, the syntax is:
+      `<revision>:<source/target name>`, any part can be omitted.
+      - Default revision is the working tree of the project
+      - Default target is the compiled project
+
+    - dataset revpath - a path to a dataset in the following format:
+      `<dataset path>:<format>`
+      - Format is optional. If not specified, will try to autodetect
+
+    - full revpath - a path to a source revision in a project, the syntax is:
+      `<project path>@<revision>:<target name>`, any part can be omitted.
+      - Default project is the current project (`-p`/`--project` CLI arg.)
+      - Default revision is the working tree of the project
+      - Default target is the compiled project
```

Comment:
I'm quite confused by this whole section.

First, the top-level list item text is "Dataset path concepts", but the anchor for it is `revpath`. The two don't correspond to each other.

Second, the second-level item promises to explain terms "source path", "dataset path", "revision path" and "project path", but the actual terms that are explained are "project local revision path", "dataset revision path" and "full revision path". That's not even the same number of terms.

Reply:
I updated the section. Probably, we lack of a dedicated term for exact "dataset path", which is only a single case of "dataset revpath".